tests/ha/control_pod/test_control_pod_soft_failures.py

```diff
@@ -0,0 +1,478 @@
+
+        LOGGER.info("Step 6: Create new IAM user and perform IOs")
+        self.test_prefix = f'test-45499-1'
+        new_user = self.mgnt_ops.create_account_users(nusers=1)
+        self.s3_clean = new_user
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(new_user.values())[0],
+                                                    log_prefix=self.test_prefix, nclients=2,
+                                                    nsamples=2)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 6: Created new IAM user and performed IOs")
+
+        LOGGER.info("ENDED: Test IOs and IAM user CRUDs before and after csm-agent is down in all "
+                    "control pods")
+
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-45500")
+    def test_iam_cruds_ios_during_soft_failure_all_pods(self):
+        """
+        Test IOs and IAM user CRUDs during soft failures in all control pods (Negative Scenario)
+        """
+        LOGGER.info("STARTED: Test IOs and IAM user CRUDs during soft failures in all control pods"
+                    " (Negative Scenario)")
+
+        event = threading.Event()
+        iam_output = Queue()
+        bkt_output = Queue()
+        num_users = HA_CFG["s3_operation_data"]["iam_users"]
+        num_bkts = HA_CFG["s3_operation_data"]["no_bkt_del_ctrl_pod"]
+        proc_restart_delay = HA_CFG["common_params"]["min_control_proc_restart_delay"]
+        LOGGER.info("Scale replicas for control pod to %s", self.repl_num)
+        resp = self.node_master_list[0].create_pod_replicas(num_replica=self.repl_num,
+                                                            deploy=const.CONTROL_POD_NAME_PREFIX)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Total number of control pods in the cluster are: %s", self.repl_num)
+
+        LOGGER.info("Adding %s delay in config files of all control pods", proc_restart_delay)
+        self.add_proc_restart_delay(proc_restart_delay)
+        LOGGER.info("Successfully added %s delay in config files of all control pods",
+                    proc_restart_delay)
+        self.change_proc_delay = True
+
+        LOGGER.info("Step 1: Perform WRITE-READ-Verify on user %s", self.uids[0])
+        self.test_prefix = 'test-45500'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(self.users.values())[0],
+                                                    log_prefix=self.test_prefix, skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed WRITEs-READs-Verify with variable sizes objects.")
+
+        LOGGER.info("Create %s iam users for deletion", num_users)
+        del_users = self.mgnt_ops.create_account_users(nusers=num_users)
+
+        LOGGER.info("Step 2: Perform IAM user creation/deletion in background")
+        args = {'user_crud': True, 'bkt_crud': False, 'num_users': num_users, 's3_obj': self.s3_obj,
```

Comment:
'bkt_crud': False is by default False, not needed
s3_obj': self.s3_obj also not needed as we are not performing any bucket CRUD.